### PR TITLE
update map

### PR DESCRIPTION
**Pull request by Daolq**
update map

### CODE REVIEW

File: app/src/main/java/vn/asiantech/asiantechfamily/activities/MainActivity.java [hunk 1]
```diff
@@ -229,9 +229,8 @@ public boolean onOptionsItemSelected(MenuItem item) {
     protected void onActivityResult(int requestCode, int resultCode, Intent data) {
         super.onActivityResult(requestCode, resultCode, data);
         if (resultCode == RESULT_OK) {
-            switch (requestCode) {
-                case NearMeFragment.REQUEST_CHECK_SETTINGS:
-                    NearMeFragment.displayViewsMap(this);
+            if(requestCode == NearMeFragment.REQUEST_CHECK_SETTINGS){
```
Comment (tranminhlan) [inline app/src/main/java/vn/asiantech/asiantechfamily/activities/MainActivity.java]: ERROR: 'if' is not followed by whitespace.

Comment (tranminhlan) [inline app/src/main/java/vn/asiantech/asiantechfamily/activities/MainActivity.java]: ERROR: '{' is not preceded with whitespace.

Comment (tranminhlan) [inline app/src/main/java/vn/asiantech/asiantechfamily/activities/MainActivity.java]: ERROR: Must have at least one statement.

File: app/src/main/java/vn/asiantech/asiantechfamily/fragments/NearMeFragment.java [hunk 2]
```diff
@@ -48,31 +50,37 @@
 @EFragment(R.layout.fragment_near_me)
 public class NearMeFragment extends BaseFragment implements
         GoogleMap.OnMarkerClickListener,
+        Animation.AnimationListener,
+        GoogleMap.OnMapClickListener,
         OnMapReadyCallback,
-        ResultCallback<LocationSettingsResult> {
+        ResultCallback<LocationSettingsResult>{
```
Comment (tranminhlan) [inline app/src/main/java/vn/asiantech/asiantechfamily/fragments/NearMeFragment.java]: ERROR: '>' is followed by an illegal character.

Comment (tranminhlan) [inline app/src/main/java/vn/asiantech/asiantechfamily/fragments/NearMeFragment.java]: ERROR: '{' is not preceded with whitespace.

File: app/src/main/java/vn/asiantech/asiantechfamily/fragments/NearMeFragment.java [hunk 3]
```diff
@@ -156,9 +188,12 @@ public void onMapReady(GoogleMap googleMap) {
 
     @Override
     public boolean onMarkerClick(Marker marker) {
-        sMap.animateCamera(CameraUpdateFactory.newLatLng(marker.getPosition()));
+        mMap.animateCamera(CameraUpdateFactory.newLatLng(marker.getPosition()));
         marker.showInfoWindow();
-        displayInfor(marker);
+        if(marker.getTitle()!=null){
```
Comment (tranminhlan) [inline app/src/main/java/vn/asiantech/asiantechfamily/fragments/NearMeFragment.java]: ERROR: 'if' is not followed by whitespace.

Comment (tranminhlan) [inline app/src/main/java/vn/asiantech/asiantechfamily/fragments/NearMeFragment.java]: ERROR: '!=' is not preceded with whitespace.

Comment (tranminhlan) [inline app/src/main/java/vn/asiantech/asiantechfamily/fragments/NearMeFragment.java]: ERROR: '!=' is not followed by whitespace.

Comment (tranminhlan) [inline app/src/main/java/vn/asiantech/asiantechfamily/fragments/NearMeFragment.java]: ERROR: '{' is not preceded with whitespace.

File: app/src/main/java/vn/asiantech/asiantechfamily/fragments/NearMeFragment.java [hunk 4]
```diff
@@ -260,4 +268,30 @@ private String loadJSONFromAsset() {
...
+    }
+
+    @Override
+    public void onAnimationStart(Animation animation) {
+
+    }
+
+    @Override
+    public void onAnimationEnd(Animation animation) {
+        if(animation == animationBottomtoTop) {
```
Comment (tranminhlan) [inline app/src/main/java/vn/asiantech/asiantechfamily/fragments/NearMeFragment.java]: ERROR: 'if' is not followed by whitespace.

File: app/src/main/java/vn/asiantech/asiantechfamily/fragments/NearMeFragment.java [hunk 5]
```diff
@@ -260,4 +268,30 @@ private String loadJSONFromAsset() {
...
+            rlContainInforMarker.setVisibility(View.VISIBLE);
+        }else if (animation == animationToptoBottom){
```
Comment (tranminhlan) [inline app/src/main/java/vn/asiantech/asiantechfamily/fragments/NearMeFragment.java]: ERROR: 'else' is not preceded with whitespace.

Comment (tranminhlan) [inline app/src/main/java/vn/asiantech/asiantechfamily/fragments/NearMeFragment.java]: ERROR: '}' is not followed by whitespace.

Comment (tranminhlan) [inline app/src/main/java/vn/asiantech/asiantechfamily/fragments/NearMeFragment.java]: ERROR: '{' is not preceded with whitespace.

File: app/src/main/java/vn/asiantech/asiantechfamily/models/Person.java [hunk 6]
```diff
@@ -16,10 +25,55 @@
...
-        return new LatLng(Double.parseDouble(getLatitude()),Double.parseDouble(getLongtitude()));
+        return new LatLng(Double.parseDouble(getLatitude()), Double.parseDouble(getLongtitude()));
+    }
+
+    public BitmapDescriptor getIcon() {
+        return icon;
+    }
+
+    // set icon from url
+    public void setIcon(Context mContext) {
```
Comment (tranminhlan) [inline app/src/main/java/vn/asiantech/asiantechfamily/models/Person.java]: ERROR: 'mContext' hides a field.

File: app/src/main/java/vn/asiantech/asiantechfamily/models/Person.java [hunk 7]
```diff
@@ -16,10 +25,55 @@
...
+        if (this.mContext == null) {
+            this.mContext = mContext;
+        }
+        new loadIcon().execute(getUrlprofile());
+    }
+
+    // load icon from url
+    private class loadIcon extends AsyncTask<String, Void, Bitmap> {
```
Comment (tranminhlan) [inline app/src/main/java/vn/asiantech/asiantechfamily/models/Person.java]: ERROR: Missing a Javadoc comment.

Comment (tranminhlan) [inline app/src/main/java/vn/asiantech/asiantechfamily/models/Person.java]: ERROR: Name 'loadIcon' must match pattern '^[A-Z][_a-zA-Z0-9]*$'.

File: app/src/main/java/vn/asiantech/asiantechfamily/views/PersonMarker.java [hunk 8]
```diff
@@ -15,6 +18,9 @@
  * Created by sun on 12/08/2015.
  */
 public class PersonMarker extends DefaultClusterRenderer<Person> {
+
+    private final WeakHashMap<Marker,Person> personWeakHashMap = new WeakHashMap<>();
```
Comment (tranminhlan) [inline app/src/main/java/vn/asiantech/asiantechfamily/views/PersonMarker.java]: ERROR: ',' is not followed by whitespace.

File: app/src/main/java/vn/asiantech/asiantechfamily/views/PersonMarker.java [hunk 9]
```diff
@@ -23,6 +29,20 @@ public PersonMarker(Context context, GoogleMap map, ClusterManager<Person> clust
...
+            markerOptions.position(item.getPosition()).title(item.getName()).snippet(item.getInfor()).icon(BitmapDescriptorFactory.fromResource(R.mipmap.ic_launcher));
+        } else {
+            markerOptions.position(item.getPosition()).title(item.getName()).snippet(item.getInfor()).icon(item.getIcon());
+        }
+    }
+
+    @Override
+    protected void onClusterItemRendered(Person clusterItem, Marker marker) {
+        super.onClusterItemRendered(clusterItem, marker);
+        personWeakHashMap.put(marker,clusterItem);
```
Comment (tranminhlan) [inline app/src/main/java/vn/asiantech/asiantechfamily/views/PersonMarker.java]: ERROR: ',' is not followed by whitespace.

File: app/src/main/java/vn/asiantech/asiantechfamily/views/PersonMarker.java [hunk 10]
```diff
@@ -23,6 +29,20 @@ public PersonMarker(Context context, GoogleMap map, ClusterManager<Person> clust
...
+    }
+
+    public Person getPerSonFromMarker(Marker marker){
```
Comment (tranminhlan) [inline app/src/main/java/vn/asiantech/asiantechfamily/views/PersonMarker.java]: ERROR: '{' is not preceded with whitespace.

File: app/src/main/java/vn/asiantech/asiantechfamily/models/Person.java (hunk 7 above)
Comment (Daolq) [inline app/src/main/java/vn/asiantech/asiantechfamily/models/Person.java]: I - I là  chữ i hoa anh.

**Pull request merged**